sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/UnsafeRowWriter.java

```diff
@@ -51,6 +51,7 @@ public UnsafeRowWriter(BufferHolder holder, int numFields) {
     this.nullBitsSize = UnsafeRow.calculateBitSetWidthInBytes(numFields);
     this.fixedSize = nullBitsSize + 8 * numFields;
     this.startingOffset = holder.cursor;
+    holder.reset();
```

Comment:
I not very sure about this, but what if this writer is for inner struct? Then the buffer holder is shared between many writers and we should only reset once.